onnxruntime/python/tools/quantization/test/test_calibrate.py

```diff
@@ -7,130 +7,349 @@
 # --------------------------------------------------------------------------
 
 import unittest
-import os
 import onnx
 import onnxruntime
-from onnx import helper, TensorProto
-from onnx.helper import make_node, make_tensor_value_info
-import calibrate
 import numpy as np
-from onnx import numpy_helper
+from onnx import helper, TensorProto, numpy_helper
+from onnxruntime.quantization.calibrate import calibrate, CalibrationDataReader, ONNXCalibrater
+
+
+def generate_input_initializer(tensor_shape,tensor_dtype,input_name):
+  '''
+  Helper function to generate initializers for test inputs
+  '''
+  tensor = np.random.ranf(tensor_shape).astype(tensor_dtype)
+  init = numpy_helper.from_array(tensor,input_name)
+  return init  
+
+class TestDataReader(CalibrationDataReader):
+    '''for test purpose'''
+    def __init__(self, calibration_image_folder):
+        self.image_folder = calibration_image_folder
+
+    def get_next(self):
+        return None
+
+class TestDataReaderSecond(CalibrationDataReader):
+    '''for test purpose'''
+    def __init__(self, calibration_image_folder):
+        self.image_folder = calibration_image_folder
+        self.preprocess_flag = True
+        self.enum_data_dicts = []
+        self.datasize = 3
```

Comment:
>self.datasize = 3 [](start = 8, length = 17)

Please take a look at this PR. We don't need datasize but need to fix the bug in the calibrate